docker/jenkins2-centos/Dockerfile

```diff
@@ -1,10 +1,28 @@
 FROM openshift/jenkins-2-centos7
 
-COPY ./contrib/kube-slave-common.sh /usr/local/bin/kube-slave-common.sh
+USER root
+
+# rename the run script from base image so that we can use it
+RUN mv /usr/libexec/s2i/run /usr/libexec/s2i/run-base
+
+# copy our custom Jenkins scripts
+COPY ./contrib/jenkins /usr/local/bin
+
+# copy the s2i scripts. s2i expects them to be in /usr/libexec/s2i by default.
+ADD ./contrib/s2i /usr/libexec/s2i
+
+# make the copied s2i scripts executable
+RUN chmod -R +x /usr/libexec/s2i/*
 
 COPY plugins.txt /opt/openshift/configuration/plugins.txt
-RUN /usr/local/bin/install-plugins.sh /opt/openshift/configuration/plugins.txt && \
-    mkdir $JENKINS_HOME/secrets && \
+
+USER 1001
+
+RUN /usr/local/bin/install-plugins.sh /opt/openshift/configuration/plugins.txt
+
+RUN mkdir $JENKINS_HOME/secrets && \
```

Comment:
Why splitting this? We need to minimize number of layers. By grouping (with common sense) `RUN` and `COPY` commands. Good example of clean and neat Dockerfile: https://github.com/sclorg/s2i-base-container/blob/master/Dockerfile